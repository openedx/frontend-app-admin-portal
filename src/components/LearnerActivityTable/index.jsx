import React from 'react';
import PropTypes from 'prop-types';

import TableContainer from '../../containers/TableContainer';
import { formatTimestamp, formatPassedTimestamp, formatPercentage } from '../../utils';
import EnterpriseDataApiService from '../../data/services/EnterpriseDataApiService';

class LearnerActivityTable extends React.Component {
  getTableColumns() {
    const { activity } = this.props;
    const tableColumns = [
      {
        label: 'Email',
        key: 'user_email',
        columnSortable: true,
      },
      {
        label: 'Course Title',
        key: 'course_title',
        columnSortable: true,
      },
      {
        label: 'Course Price',
        key: 'course_list_price',
        columnSortable: true,
      },
      {
        label: 'Start Date',
        key: 'course_start_date',
        columnSortable: true,
      },
      {
        label: 'End Date',
        key: 'course_end_date',
        columnSortable: true,
      },
      {
        label: 'Passed Date',
        key: 'passed_date',
        columnSortable: true,
      },
      {
        label: 'Current Grade',
        key: 'current_grade',
        columnSortable: true,
      },
      {
        label: 'Progress Status',
        key: 'progress_status',
        columnSortable: true,
      },
      {
        label: 'Last Activity Date',
        key: 'last_activity_date',
        columnSortable: true,
      },
    ];

    if (activity !== 'active_past_week') {
      return tableColumns.filter(column => column.key !== 'passed_date');
    }
    return tableColumns;
  }

  formatTableData = enrollments => enrollments.map(enrollment => ({
    ...enrollment,
    user_email: <span data-hj-suppress>{enrollment.user_email}</span>,
    last_activity_date: formatTimestamp({ timestamp: enrollment.last_activity_date }),
    course_start_date: formatTimestamp({ timestamp: enrollment.course_start_date }),
    course_end_date: formatTimestamp({ timestamp: enrollment.course_end_date }),
    enrollment_date: formatTimestamp({
      timestamp: enrollment.enrollment_date,
    }),
    passed_date: formatPassedTimestamp(enrollment.passed_date),
    user_account_creation_date: formatTimestamp({
      timestamp: enrollment.user_account_creation_date,
    }),
    progress_status: enrollment.progress_status,
    course_list_price: enrollment.course_list_price ? `$${enrollment.course_list_price}` : '',
    current_grade: formatPercentage({ decimal: enrollment.current_grade }),
  }));

  render() {
    const { activity, id } = this.props;

    return (
      <TableContainer
        id={id}
        className={id}
        key={id}
        fetchMethod={(enterpriseId, options) => EnterpriseDataApiService.fetchCourseEnrollments(
          enterpriseId,
          {
            learnerActivity: activity,
            ...options,
          },
        )}
        columns={this.getTableColumns()}
        formatData={this.formatTableData}
        tableSortable
      />
    );
  }
}

LearnerActivityTable.propTypes = {
  id: PropTypes.string.isRequired,
  activity: PropTypes.string.isRequired,
};

export default LearnerActivityTable;
