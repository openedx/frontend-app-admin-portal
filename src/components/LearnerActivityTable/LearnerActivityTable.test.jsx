import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import LearnerActivityTable from '.';

const enterpriseId = 'test-enterprise';
const mockStore = configureMockStore([thunk]);
const learnerActivityEmptyStore = mockStore({
  portalConfiguration: {
    enterpriseId,
  },
  table: {
    'active-week': {
      data: {
        results: [],
        current_page: 1,
        num_pages: 1,
      },
      ordering: null,
      loading: false,
      error: null,
    },
  },
});

const tableMockData = {
  data: {
    count: 2,
    num_pages: 1,
    current_page: 1,
    results: [
      {
        id: 1,
        passed_date: '2018-09-23T16:27:34.690065Z',
        course_title: 'Dive into ReactJS',
        course_key: 'edX/ReactJS',
        user_email: 'awesome.me@example.com',
        course_list_price: '200',
        course_start_date: '2017-10-21T23:47:32.738Z',
        course_end_date: '2018-05-13T12:47:27.534Z',
        current_grade: '0.66',
        progress_status: 'Failed',
        last_activity_date: '2018-09-22T10:59:28.628Z',
      },
      {
        id: 5,
        passed_date: '2018-09-22T16:27:34.690065Z',
        course_title: 'Redux with ReactJS',
        course_key: 'edX/Redux_ReactJS',
        user_email: 'new@example.com',
        course_list_price: '200',
        course_start_date: '2017-10-21T23:47:32.738Z',
        course_end_date: '2018-05-13T12:47:27.534Z',
        current_grade: '0.80',
        progress_status: 'Passed',
        last_activity_date: '2018-09-25T10:59:28.628Z',
      },
    ],
    next: null,
    start: 0,
    previous: null,
  },
  ordering: null,
  loading: false,
  error: null,
};

const learnerActivityStore = mockStore({
  portalConfiguration: {
    enterpriseId,
  },
  table: {
    'active-week': tableMockData,
    'inactive-week': tableMockData,
    'inactive-month': tableMockData,
  },
});

const LearnerActivityEmptyTableWrapper = props => (
  <MemoryRouter>
    <Provider store={learnerActivityEmptyStore}>
      <LearnerActivityTable
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

const LearnerActivityTableWrapper = props => (
  <MemoryRouter>
    <Provider store={learnerActivityStore}>
      <LearnerActivityTable
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

const verifyLearnerActivityTableRendered = (tableId, activity, columnTitles, rowsData) => {
  const wrapper = render((
    <LearnerActivityTableWrapper id={tableId} activity={activity} />
  ));
  // Verify that table has correct number of columns
  expect(wrapper.container.querySelectorAll(`.${tableId} thead th`).length).toEqual(columnTitles.length);

  // Verify only expected columns are shown
  wrapper.container.querySelectorAll(`.${tableId} thead th`).forEach((column, index) => {
    expect(column.textContent).toContain(columnTitles[index]);
  });

  // Verify that table has correct number of rows
  expect(wrapper.container.querySelectorAll(`.${tableId} tbody tr`).length).toEqual(2);

  // Verify each row in table has correct data
  wrapper.container.querySelectorAll(`.${tableId} tbody tr`).forEach((row, rowIndex) => {
    row.querySelectorAll('td').forEach((cell, colIndex) => {
      expect(cell.textContent).toEqual(rowsData[rowIndex][colIndex]);
    });
  });
};

describe('LearnerActivityTable', () => {
  it('renders empty state correctly', () => {
    const { container: tree } = render(
      <LearnerActivityEmptyTableWrapper id="active-week" activity="active_past_week" />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders active learners table correctly', () => {
    const { container: tree } = render(
      <LearnerActivityTableWrapper id="active-week" activity="active_past_week" />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders inactive past week learners table correctly', () => {
    const { container: tree } = render(
      <LearnerActivityTableWrapper id="inactive-week" activity="inactive_past_week" />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders inactive past month learners table correctly', () => {
    const { container: tree } = render(
      <LearnerActivityTableWrapper id="inactive-month" activity="inactive_past_month" />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders active learners table with correct data', () => {
    const tableId = 'active-week';
    const activity = 'active_past_week';
    const columnTitles = [
      'Email',
      'Course Title',
      'Course Price',
      'Start Date',
      'End Date',
      'Passed Date',
      'Current Grade',
      'Progress Status',
      'Last Activity Date',
    ];
    const rowsData = [
      [
        'awesome.me@example.com',
        'Dive into ReactJS',
        '$200',
        'October 21, 2017',
        'May 13, 2018',
        'September 23, 2018',
        '66%',
        'Failed',
        'September 22, 2018',
      ],
      [
        'new@example.com',
        'Redux with ReactJS',
        '$200',
        'October 21, 2017',
        'May 13, 2018',
        'September 22, 2018',
        '80%',
        'Passed',
        'September 25, 2018',
      ],
    ];

    verifyLearnerActivityTableRendered(tableId, activity, columnTitles, rowsData);
  });

  it('renders inactive past week learners table with correct data', () => {
    const tableId = 'inactive-week';
    const activity = 'inactive_past_week';
    const columnTitles = [
      'Email',
      'Course Title',
      'Course Price',
      'Start Date',
      'End Date',
      'Current Grade',
      'Progress Status',
      'Last Activity Date',
    ];
    const rowsData = [
      [
        'awesome.me@example.com',
        'Dive into ReactJS',
        '$200',
        'October 21, 2017',
        'May 13, 2018',
        '66%',
        'Failed',
        'September 22, 2018',
      ],
      [
        'new@example.com',
        'Redux with ReactJS',
        '$200',
        'October 21, 2017',
        'May 13, 2018',
        '80%',
        'Passed',
        'September 25, 2018',
      ],
    ];

    verifyLearnerActivityTableRendered(tableId, activity, columnTitles, rowsData);
  });

  it('renders inactive past month learners table with correct data', () => {
    const tableId = 'inactive-month';
    const activity = 'inactive_past_month';
    const columnTitles = [
      'Email',
      'Course Title',
      'Course Price',
      'Start Date',
      'End Date',
      'Current Grade',
      'Progress Status',
      'Last Activity Date',
    ];
    const rowsData = [
      [
        'awesome.me@example.com',
        'Dive into ReactJS',
        '$200',
        'October 21, 2017',
        'May 13, 2018',
        '66%',
        'Failed',
        'September 22, 2018',
      ],
      [
        'new@example.com',
        'Redux with ReactJS',
        '$200',
        'October 21, 2017',
        'May 13, 2018',
        '80%',
        'Passed',
        'September 25, 2018',
      ],
    ];

    verifyLearnerActivityTableRendered(tableId, activity, columnTitles, rowsData);
  });
});
