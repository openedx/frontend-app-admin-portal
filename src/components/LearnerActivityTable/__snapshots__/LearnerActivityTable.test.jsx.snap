// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnerActivityTable renders active learners table correctly 1`] = `
<div
  className="active-week"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table table-sm table-striped"
        >
          <thead
            className=""
          >
            <tr
              className=""
            >
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Email
                    <span
                      className="sr-only"
                    >
                       
                      sort ascending
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort-asc"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Course Title
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Course Price
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Start Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    End Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Passed Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Current Grade
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Progress Status
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Last Activity Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody
            className=""
          >
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  awesome.me@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Dive into ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                $200
              </td>
              <td
                className=""
                scope={null}
              >
                October 21, 2017
              </td>
              <td
                className=""
                scope={null}
              >
                May 13, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                September 23, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                66%
              </td>
              <td
                className=""
                scope={null}
              >
                Failed
              </td>
              <td
                className=""
                scope={null}
              >
                September 22, 2018
              </td>
            </tr>
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  new@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Redux with ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                $200
              </td>
              <td
                className=""
                scope={null}
              >
                October 21, 2017
              </td>
              <td
                className=""
                scope={null}
              >
                May 13, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                September 22, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                80%
              </td>
              <td
                className=""
                scope={null}
              >
                Passed
              </td>
              <td
                className=""
                scope={null}
              >
                September 25, 2018
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="row mt-2"
  >
    <div
      className="col d-flex justify-content-center"
    >
      <nav
        aria-label="active-week-pagination"
        className="pagination-default"
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          aria-relevant="text"
          className="sr-only"
        >
          Page 1, Current Page, of 1
        </div>
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Previous"
              className="previous page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12l4.58-4.59Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
                Previous
              </div>
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Next"
              className="next page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                Next
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6-6-6Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`LearnerActivityTable renders empty state correctly 1`] = `
<div
  className="fade alert-content alert alert-warning show"
  role="alert"
>
  <span
    className="pgn__icon alert-icon"
  >
    <svg
      aria-hidden={true}
      fill="none"
      focusable={false}
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm1 15h-2v-2h2v2Zm0-4h-2V7h2v6Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <div
    className="pgn__alert-message-wrapper"
  >
    <div
      className="alert-message-content"
    >
      There are no results.
    </div>
  </div>
</div>
`;

exports[`LearnerActivityTable renders inactive past month learners table correctly 1`] = `
<div
  className="inactive-month"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table table-sm table-striped"
        >
          <thead
            className=""
          >
            <tr
              className=""
            >
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Email
                    <span
                      className="sr-only"
                    >
                       
                      sort ascending
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort-asc"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Course Title
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Course Price
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Start Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    End Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Current Grade
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Progress Status
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Last Activity Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody
            className=""
          >
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  awesome.me@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Dive into ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                $200
              </td>
              <td
                className=""
                scope={null}
              >
                October 21, 2017
              </td>
              <td
                className=""
                scope={null}
              >
                May 13, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                66%
              </td>
              <td
                className=""
                scope={null}
              >
                Failed
              </td>
              <td
                className=""
                scope={null}
              >
                September 22, 2018
              </td>
            </tr>
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  new@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Redux with ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                $200
              </td>
              <td
                className=""
                scope={null}
              >
                October 21, 2017
              </td>
              <td
                className=""
                scope={null}
              >
                May 13, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                80%
              </td>
              <td
                className=""
                scope={null}
              >
                Passed
              </td>
              <td
                className=""
                scope={null}
              >
                September 25, 2018
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="row mt-2"
  >
    <div
      className="col d-flex justify-content-center"
    >
      <nav
        aria-label="inactive-month-pagination"
        className="pagination-default"
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          aria-relevant="text"
          className="sr-only"
        >
          Page 1, Current Page, of 1
        </div>
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Previous"
              className="previous page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12l4.58-4.59Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
                Previous
              </div>
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Next"
              className="next page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                Next
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6-6-6Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`LearnerActivityTable renders inactive past week learners table correctly 1`] = `
<div
  className="inactive-week"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table table-sm table-striped"
        >
          <thead
            className=""
          >
            <tr
              className=""
            >
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Email
                    <span
                      className="sr-only"
                    >
                       
                      sort ascending
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort-asc"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Course Title
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Course Price
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Start Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    End Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Current Grade
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Progress Status
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Last Activity Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody
            className=""
          >
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  awesome.me@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Dive into ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                $200
              </td>
              <td
                className=""
                scope={null}
              >
                October 21, 2017
              </td>
              <td
                className=""
                scope={null}
              >
                May 13, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                66%
              </td>
              <td
                className=""
                scope={null}
              >
                Failed
              </td>
              <td
                className=""
                scope={null}
              >
                September 22, 2018
              </td>
            </tr>
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  new@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Redux with ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                $200
              </td>
              <td
                className=""
                scope={null}
              >
                October 21, 2017
              </td>
              <td
                className=""
                scope={null}
              >
                May 13, 2018
              </td>
              <td
                className=""
                scope={null}
              >
                80%
              </td>
              <td
                className=""
                scope={null}
              >
                Passed
              </td>
              <td
                className=""
                scope={null}
              >
                September 25, 2018
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="row mt-2"
  >
    <div
      className="col d-flex justify-content-center"
    >
      <nav
        aria-label="inactive-week-pagination"
        className="pagination-default"
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          aria-relevant="text"
          className="sr-only"
        >
          Page 1, Current Page, of 1
        </div>
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Previous"
              className="previous page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12l4.58-4.59Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
                Previous
              </div>
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Next"
              className="next page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                Next
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6-6-6Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
